pkg/segment/context/context.go

```diff
@@ -0,0 +1,61 @@
+package context
+
+import (
+	"context"
+	"sync"
+)
+
+type contextKey struct{}
+
+var key = contextKey{}
+
+type properties struct {
+	lock    sync.Mutex
+	storage map[string]interface{}
+}
+
+func (p *properties) set(name string, value interface{}) {
```

Comment:
To be honest, a lot of this code is copied from another project([crc](https://github.com/code-ready/crc/blob/master/pkg/crc/telemetry)), I modified it for our usage, most of the things are unchanged, so if you think something needs to be changed, let me know. I let these methods be private because I don't intend to use them outside of this file.

Reply:
Yes, it is better to keep these methods private.

The only thing I would change would be to place the public methods/functions before the private ones in the file.

Reply:
Ah right, I misread your comment the first time :facepalm: